framework/src/kernels/NonlocalKernel.C

```diff
@@ -39,6 +39,58 @@ NonlocalKernel::NonlocalKernel(const InputParameters & parameters) :
 }
 
 void
+NonlocalKernel::computeJacobian()
+{
```

Comment:
At the very least could you put documentation in each of these methods letting maintainers know that this code is copied from Kernel. That way if somebody comes in here and makes a change, they can check the other implementation as well.